.github/workflows/test.yaml

```diff
@@ -0,0 +1,46 @@
+name: Tests
+
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [windows-2019, windows-latest]
+    
+    runs-on: ${{ matrix.platform }}
+
+    steps:
+    # Environment requirements:
+    # https://learn.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment
+    - name: Test docker
+      run: |
+        docker version
+        docker pull hello-world
+        docker run hello-world
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: 'stable'
+    
+    - name: Install Dependencies
+      run: |
+        go install github.com/magefile/mage@latest
+        go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2
+
+    - name: Run E2E tests
+      shell: pwsh
+      run: |
+        Install-Module -Name DockerMsftProvider -Force
```

Comment:
Are the `Install-Module` and `Import-Module` statements required for this to work? My understanding was that both the windows 2019 and 2022 github runners came preconfigured with docker ([2019 details](https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md#tools), [2022 details](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#tools)). Also, the `DockerMsftProvider` [module is deprecated](https://github.com/OneGet/MicrosoftDockerProvider?tab=readme-ov-file) (But for whatever reason still seems to work).

Reply:
You are absolutely right, both 2019 and 2022 GHA runners come with Docker pre-installed.

The reason why I added this `Install-Module` and `Import-Module` was to be able to get [hns_test.ps1](https://github.com/rancher/wins/blob/main/tests/integration/hns_test.ps1) to run. Without this it fails with `Get-HnsNetwork` is to recognised IIRC.

Reply:
Interesting, I thought the hns utility commands would be bundled together with docker support. I guess leaving the `DockerMsftProvider` line can't hurt, if it fails to install docker then we will just use what is already installed. The `Import-Module` line makes more sense now